libbeat/monitoring/report/elasticsearch/client.go

```diff
@@ -150,3 +145,45 @@ func (c *publishClient) Test(d testing.Driver) {
 func (c *publishClient) String() string {
 	return "publish(" + c.es.String() + ")"
 }
+
+func (c *publishClient) bulkToProduction(params map[string]string, event publisher.Event, docType interface{}) error {
+	action := common.MapStr{
+		"index": common.MapStr{
+			"_type":    docType,
+			"_index":   "",
+			"_routing": nil,
+		},
+	}
+	document := report.Event{
+		Timestamp: event.Content.Timestamp,
+		Fields:    event.Content.Fields,
+	}
+	bulk := [2]interface{}{action, document}
+
+	// Currently one request per event is sent. Reason is that each event can contain different
+	// interval params and X-Pack requires to send the interval param.
+	_, err := c.es.BulkWith("_xpack", "monitoring", params, nil, bulk[:])
+	return err
+}
+
+// TODO: figure out why this isn't actually indexing anything!
+func (c *publishClient) bulkToMonitoring(event publisher.Event) error {
+	action := common.MapStr{
+		"index": common.MapStr{
+			"_type":    "doc",
```

Comment:
this one is annoying. If the target ES version is 7.0 we will have to remove the `_type` field.